paddle/operators/chunk_eval_op.cc

```diff
@@ -0,0 +1,140 @@
+      : OpProtoAndCheckerMaker(proto, op_checker) {
+    AddInput("Inference",
+             "(Tensor, default: Tensor<int>) Predictions from the network.");
+    AddInput("Label", "(Tensor, default: Tensor<int>) Labels of the data.");
+    AddOutput(
+        "Precision",
+        "(float) The precision ratio of the predictions on current data.");
+    AddOutput("Recall",
+              "(float) The recall ratio of the predictions on current data.");
+    AddOutput("F1-Score",
+              "(float) The F1-Score of the predictions on current data.");
+    AddAttr<int>("num_chunk_types", "(int) The number of chunk type.");
+    AddAttr<std::string>("chunk_scheme",
+                         "(string, default IOB) The label scheme.")
+        .SetDefault("IOB");
+    AddAttr<std::vector<int>>(
+        "excluded_chunk_types",
+        "(list<int>) A list<int> indicating chunk types not to be counted.")
+        .SetDefault(std::vector<int>{});
+    AddComment(R"DOC(
+Chunk evaluator is used to evaluate segment labelling accuracy for a
+sequence. It calculates precision, recall and F1 scores for the chunk detection.
+To use chunk evaluator, several concepts need to be clarified firstly.
+[Chunk type] is the type of the whole chunk and a chunk consists of one or several words.  (For example in NER, ORG for organization name, PER for person name etc.)
+[Tag type] indicates the position of a word in a chunk. (B for begin, I for inside, E for end, S for single)
+We can name a label by combining tag type and chunk type. (ie. B-ORG for begining of an organization name)
+The construction of label dictionary should obey the following rules:
+- Use one of the listed labelling schemes. These schemes differ in ways indicating chunk boundry.
+
+    Scheme    Description
+    plain    Use the same label for the whole chunk.
+    IOB      Two labels for chunk type X, B-X for chunk begining and I-X for chunk inside.
+    IOE      Two labels for chunk type X, E-X for chunk ending and I-X for chunk inside.
+    IOBES    Four labels for chunk type X, B-X for chunk begining, I-X for chunk inside, E-X for chunk end and S-X for single word chunk.
+
+To make it clear, let's illustrate by an NER example.
+Assuming that there are three named entity types including ORG, PER and LOC which are called 'chunk type' here,
+if 'IOB' scheme were used, the label set will be extended to a set including B-ORG, I-ORG, B-PER, I-PER, B-LOC, I-LOC and O,
+in which B-ORG for begining of ORG and I-ORG for inside of ORG.
+Prefixes which are called 'tag type' here are added to chunk types and there are two tag types including B and I.
+Of course, the training data should be labeled accordingly.
+- Mapping is done correctly by the listed equations and assigning protocol.
+The following table are equations to extract tag type and chunk type from a label.
+
+    tagType = label % numTagType
+    chunkType = label / numTagType
+    otherChunkType = numChunkTypes
```

Comment:
The `numTagType` and `numChunkTypes` here is clear, but better to explain them again.

Reply:
Rewrite the doc.